File: src/main/java/system/search/results/CalculatorResult.java
package system.search.results;

import javafx.scene.image.Image;
import system.search.SearchEngine;

import java.math.BigDecimal;
import java.util.ResourceBundle;

public class CalculatorResult extends AbstractResult {
    // This field is used in the search bar to display the filter label
    // It refers to the resource bundle id
    public static final String SEARCH_FILDER_RESOURCE_ID = "calculator_category";

    private String expression;
    private BigDecimal result;

    public CalculatorResult(SearchEngine searchEngine, ResourceBundle resourceBundle, String query, BigDecimal result,
                            Image defaultImage) {
        super(searchEngine, resourceBundle, defaultImage);
        this.expression = query;
        this.result = result;
    }

    @Override
    public String getTitle() {
        return expression + " = "+result;
    }

    @Override
    public String getDescription() {
        return resourceBundle.getString("result_of_expression") + " " + result;
    }

    @Override
    public void requestImage(OnImageAvailableListener listener) {
        if (listener != null) {
            new Thread(() -> {
                listener.onImageAvailable(defaultImage, null);
            }).start();
        }
    }

    @Override
    public void executeAction() {

    }
}

File: src/main/java/system/search/results/CalculatorResult.kt
package system.search.results

import model.command.Command
import system.context.SearchContext
import system.search.annotations.FilterableResult
import java.math.BigDecimal

class CalculatorResult(context: SearchContext, val query : String, val result: BigDecimal) : AbstractResult(context) {
    override val title: String
        get() = "$query = $result"
    override val description: String?
        get() = "${context.resourceBundle.getString("result_of_expression")} $result"

    override val category = ResultCategory(context.resourceBundle.getString("calculator_category"), 100)

            override val imageId: String?
        get() = "asset:light"
}